examples/vision/attention_mll_classification.py

```diff
@@ -0,0 +1,579 @@
+train_data, train_labels = create_bags(x, y, POS_CLASS, BAG_COUNT, BAG_SIZE)
+
+# Create validation data.
+val_data, val_labels = create_bags(x_val, y_val, POS_CLASS, VAL_BAG_COUNT, BAG_SIZE)
+
+"""
+# Create and Train Neural Networks.
+
+We will now build the attention layer, prepare some utilities, build and train the
+entire model.
+
+## Attention Operator Implementation
+
+The output size of this layer is decided by the size of a single bag.
+
+The attention mechanism uses a weighted average of instances in a bag, in which the sum
+of the weights must equal to 1 (invariant of the bag size).
+
+The weight matrices (parameters) are **w** and **v**. To include positive and negative
+values, hyperbolic tangent element-wise non-linearity is utilized.
+
+**Gated attention mechanism** can be used to deal with complex relations. Another weight
+matrix, **u**, is added to the computation.
+Sigmoid non-linearity is used to overcome approximately linear behavior for *x* ∈ [−1, 1]
+by hyperbolic tangent non-linearity.
+"""
+
+
+class AttentionLayer(layers.Layer):
+    def __init__(
+        self,
+        l_dim,
+        output_dim,
+        kernel_initializer="glorot_uniform",
+        kernel_regularizer=None,
+        use_gated=False,
+        **kwargs,
+    ):
+        self.l_dim = l_dim
+        self.output_dim = output_dim
+        self.use_gated = use_gated
+
+        self.kernel_initializer = keras.initializers.get(kernel_initializer)
+        self.kernel_regularizer = keras.regularizers.get(kernel_regularizer)
+
+        self.v_init = self.kernel_initializer
+        self.w_init = self.kernel_initializer
+        self.u_init = self.kernel_initializer
+
+        self.v_regularizer = self.kernel_regularizer
+        self.w_regularizer = self.kernel_regularizer
+        self.u_regularizer = self.kernel_regularizer
+
+        super(AttentionLayer, self).__init__(**kwargs)
+
+    def build(self, input_shape):
+        assert isinstance(input_shape, list)
+        # Input Shape
+        # List of 2D tensors with shape: (batch_size, input_dim).
+        input_dim = input_shape[0][1]
+
+        self.v = self.add_weight(
+            shape=(input_dim, self.l_dim),
+            initializer=self.v_init,
+            name="v",
+            regularizer=self.v_regularizer,
+            trainable=True,
+        )
+
+        self.w = self.add_weight(
+            shape=(self.l_dim, 1),
+            initializer=self.w_init,
+            name="w",
+            regularizer=self.w_regularizer,
+            trainable=True,
+        )
+
+        if self.use_gated:
+            self.u = self.add_weight(
+                shape=(input_dim, self.l_dim),
+                initializer=self.u_init,
+                name="u",
+                regularizer=self.u_regularizer,
+                trainable=True,
+            )
+        else:
+            self.u = None
+
+        self.input_built = True
+
+    def call(self, x, mask=None):
+        def compute(x):
```

Comment:
Use a separate method on the class, and give it an explicit name

Reply:
Alright, will do.